### ISSUE TITLE
provide examples

### ISSUE THREAD

**Issue by softprops** [palantir/rust-zipkin#25]
I'm reading the structure of this crate to understand the pieces but I feel like I'm missing parts. It may be helpful to provide some cargo /examples application to demonstrate how these pieces fit together

**Comment by sfackler** [comment]
Sure - I think a little toy HTTP server would make sense as a decent example.

**Comment by softprops** [comment]
In particular I'm wanting to see how to transfer contexts across threads. Since CurrentGuard can't be Send, I'm missing what the pattern in practice looks like

**Comment by sfackler** [comment]
If you have an `OpenSpan`, you can use the `detach` and `attach` methods:

```rust
let span = my_tracer.next_span()
    .with_name("background-thingy")
    .detach();

thread::spawn(move || {
    let span = span.attach();
    // do stuff...
});
```

If you don't have an `OpenSpan` but just a `CurrentGuard`, you can just grab the context from it and send that to the other thread:

```rust
let context = current_guard.context();

thread::spawn(move || {
    let current_guard = tracer.set_current(context);
    // do stuff...
});
```

**Comment by softprops** [comment]
In that first example, does the span information associated with "background-thing" get discarded when detaching? It seems the both detach/attach operations discard any built up span state. When attaching, it seems all operations on the open span would have no effect. I'm I understanding this right?

**Comment by sfackler** [comment]
No, attaching and detaching doesn't discard any state. The only thing it changes is if the OpenSpan acts as a CurrentGuard in addition.

**Comment by softprops** [comment]
I'm looking to understand the use of `mem::replace` [here](https://github.com/palantir/rust-zipkin/blob/8e687536b2d95ec002599b7aaa4a71e6797867e0/zipkin/src/tracer.rs#L237-L238)

**Comment by sfackler** [comment]
Detach consumes the old OpenSpan by value an returns a new one with the span's state.

**Comment by softprops** [comment]
Ah I see. Thanks!
